src/test/java/org/isf/accounting/test/Tests.java

```diff
@@ -927,63 +555,57 @@ public void mgrGetBillsBetweenDatesWherePatient() throws OHException, OHServiceE
 	}
 
 	@Test
-	public void mgrGetPendingBillsForPatientId() throws OHException, OHServiceException {
+	public void mgrGetPendingBillsForPatientId() throws Exception {
 		int id = _setupTestBill(false);
-		Bill foundBill = (Bill)jpa.find(Bill.class, id);
+		Bill foundBill = accountingBillIoOperationRepository.findOne(id);
 		ArrayList<Bill> bills = billBrowserManager.getPendingBills(foundBill.getBillPatient().getCode());
 		assertThat(bills.get(0).getAmount()).isCloseTo(foundBill.getAmount(), offset(0.1));
 	}
 
 	@Test
-	public void mgrNewBillNoItemsNoPayments() throws OHException, OHServiceException {
+	public void mgrNewBillNoItemsNoPayments() throws Exception {
 		int billId = _setupTestBill(false);
 		int billItemsId = _setupTestBillItems(false);
 
 		Patient patient = testPatient.setup(false);
 		PriceList priceList = testPriceList.setup(false);
-		jpa.beginTransaction();
-		jpa.persist(priceList);
-		jpa.persist(patient);
-		jpa.commitTransaction();
+		priceListIoOperationRepository.saveAndFlush(priceList);
+		patientIoOperationRepository.saveAndFlush(patient);
 		Bill bill = testBill.setup(priceList, patient, false);
 		boolean success = billBrowserManager.newBill(
 				bill,
-				new ArrayList<BillItems>(),
-				new ArrayList<BillPayments>());
+				new ArrayList<>(),
+				new ArrayList<>());
 		assertThat(success).isTrue();
 	}
 
 	@Test
-	public void mgrNewBillBillItemsNoPayments() throws OHException, OHServiceException {
+	public void mgrNewBillBillItemsNoPayments() throws Exception {
 		Patient patient = testPatient.setup(false);
 		PriceList priceList = testPriceList.setup(false);
-		jpa.beginTransaction();
-		jpa.persist(priceList);
-		jpa.persist(patient);
-		jpa.commitTransaction();
+		priceListIoOperationRepository.saveAndFlush(priceList);
+		patientIoOperationRepository.saveAndFlush(patient);
 		Bill bill = testBill.setup(priceList, patient, false);
 		BillItems insertBillItem = testBillItems.setup(null, false);
 		ArrayList<BillItems> billItems = new ArrayList<>();
 		billItems.add(insertBillItem);
 		boolean success = billBrowserManager.newBill(
 				bill,
 				billItems,
-				new ArrayList<BillPayments>());
+				new ArrayList<>());
 		assertThat(success).isTrue();
 	}
 
 	@Test
-	public void mgrNewBillNoItemsAndPayments() throws OHException, OHServiceException {
+	public void mgrNewBillNoItemsAndPayments() throws Exception {
 		Patient patient = testPatient.setup(false);
 		PriceList priceList = testPriceList.setup(false);
-		jpa.beginTransaction();
-		jpa.persist(priceList);
-		jpa.persist(patient);
-		jpa.commitTransaction();
+		priceListIoOperationRepository.saveAndFlush(priceList);
```

Comment:
Not sure I understand the question; are you saying this isn't the best/safest way to perform the action?

Reply:
I think this flush is unnecessary operation and may introduce some confusion there. By default if save is executed outside transaction context JpaRepository opens new transaction and commits everything after going outside of save method. So data should be visible for next operations without flushing. And as you saw in MergePatientTest it didn't helped in case of nested transactions because in such situation saveAndFlush joins parent transaction. 

Probably this saveAndFlush might be useful in totally different situation when we don't use spring transaction management and manage them manually


 It's very minor so you can ignore this comment